### PR TITLE
Avoid duplicate entries for process candidate starters and task candidates

### LINKED ISSUE #4185: Avoid duplicate entries for process candidate starter and task candidate
Avoid duplicate entries for process candidate starter and task candidate when receiving events from RB.

### PR DESCRIPTION
**Pull request by balsarori**
fix https://github.com/Activiti/Activiti/issues/4185

### CODE REVIEW

File: activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/test/java/org/activiti/cloud/services/query/events/handlers/ProcessCandidateStarterGroupAddedEventHandlerTest.java [hunk 1]
```diff
@@ -58,6 +61,27 @@ public void handleShouldStoreNewProcessCandidateGroupUser() {
...
+        ProcessCandidateStarterGroupImpl candidateGroup = new ProcessCandidateStarterGroupImpl(UUID.randomUUID().toString(),
+                                                                                               UUID.randomUUID().toString());
+        CloudProcessCandidateStarterGroupAddedEventImpl event = new CloudProcessCandidateStarterGroupAddedEventImpl(candidateGroup);
+
+        //when
+        handler.handle(event);
+        when(entityManager.find(ProcessCandidateStarterGroupEntity.class,
+                                new ProcessCandidateStarterGroupId(candidateGroup.getProcessDefinitionId(), candidateGroup.getGroupId())))
+            .thenReturn(new ProcessCandidateStarterGroupEntity());
+        handler.handle(event);
```
Comment (erdemedeiros) [inline activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/test/java/org/activiti/cloud/services/query/events/handlers/ProcessCandidateStarterGroupAddedEventHandlerTest.java]: Suggestion: as this is a unit test and the case where handle is storing the entity is already tested in the previous test, I think you can remove the call to `handle.handle(event)` at the line `72` and keep only the one at the line `75`. But you can keep it like this if you think it's clearer on that way.

File: activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/test/java/org/activiti/cloud/services/query/events/handlers/ProcessCandidateStarterGroupAddedEventHandlerTest.java [hunk 2]
```diff
@@ -58,6 +61,27 @@ public void handleShouldStoreNewProcessCandidateGroupUser() {
...
+
+        //then
+        ArgumentCaptor<ProcessCandidateStarterGroupEntity> captor = ArgumentCaptor.forClass(ProcessCandidateStarterGroupEntity.class);
+        verify(entityManager, times(1)).persist(captor.capture());
+        assertThat(captor.getValue().getProcessDefinitionId()).isEqualTo(event.getEntity().getProcessDefinitionId());
+        assertThat(captor.getValue().getGroupId()).isEqualTo(event.getEntity().getGroupId());
```
Comment (erdemedeiros) [inline activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/test/java/org/activiti/cloud/services/query/events/handlers/ProcessCandidateStarterGroupAddedEventHandlerTest.java]: Then this bock becomes:
`verify(entityManager, never()).persist(any());`
What do you think?

Comment (balsarori) [reply activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/test/java/org/activiti/cloud/services/query/events/handlers/ProcessCandidateStarterGroupAddedEventHandlerTest.java]: makes sense. I'll update tests

**Review by erdemedeiros — APPROVED** [review]
LGTM!

**Pull request merged**